src/core/zowe/core_for_zowe_sdk/config_file.py

```diff
@@ -340,3 +345,94 @@ def load_secure_props(self) -> None:
                 f" with error '{error_msg}'",
                 SecurePropsNotFoundWarning,
             )
+
+    def _retrieve_credential(self, service_name: str) -> Optional[str]:
+        """
+        Retrieve the credential from the keyring or storage.
+        If the credential exceeds the maximum length, retrieve it in parts.
+        Parameters
+        ----------
+        service_name: str
+            The service name for the credential retrieval
+        Returns
+        -------
+        str
+            The retrieved  encoded credential
+        """
+        encoded_credential = keyring.get_password(service_name, constants["ZoweAccountName"])
+
+        if encoded_credential is None:
+            # Retrieve the secure value with an index
+            index = 1
+            temp_value = keyring.get_password(f"{service_name}-{index}", f"{constants['ZoweAccountName']}-{index}")
+            while temp_value is not None:
+                if encoded_credential is None:
+                    encoded_credential = temp_value
+                else:
+                    encoded_credential += temp_value
+                index += 1
+                temp_value = keyring.get_password(f"{service_name}-{index}", f"{constants['ZoweAccountName']}-{index}")
+                
+        if encoded_credential is not None and encoded_credential.endswith("\0"):
+            encoded_credential = encoded_credential[:-1]
+            
+        
+        return encoded_credential 
+    
+    def set_secure_props(self) -> None:
+        """
+        Set secure_props for the given config file
+        Returns
+        -------
+        None
+        """
+        if not HAS_KEYRING:
+            return
+
+        try:
+            service_name = constants["ZoweServiceName"]
+            credential = self.secure_props.get(self.filepath, {})
+            # Check if credential is a non-empty string
+            if credential:
+                
+                # Encode the credential
+                encoded_credential = base64.b64encode(commentjson.dumps(credential).encode()).decode()
+                if sys.platform == "win32":
+                    service_name += "/" + constants["ZoweAccountName"]
+                    
+                    # Load existing credentials
+                    existing_credential = self._retrieve_credential(service_name)
+                    if existing_credential:
+                        # Decode the existing credential and update secure_props
+                        existing_credential_bytes = base64.b64decode(existing_credential.encode())
+                        existing_secure_props = commentjson.loads(existing_credential_bytes)
+                        existing_secure_props[self.filepath] = self.secure_props[self.filepath]
+                        encoded_credential = base64.b64encode(commentjson.dumps(existing_secure_props).encode()).decode()
```

Comment:
This logic should be moved up outside of the `if sys.platform == "win32"` condition, because on all platforms we want to merge credentials for this file into the credentials object that may already exist in the keyring.